### PR TITLE
Add MAC F64 Instruction to Support gfx11/gfx12

### PR DESCRIPTION
**Pull request by wenchuanchen**
Add f64 mac instruction for tensilelite and f64 tuning yamls as well. Also add testing yamls including f32 and f64 for tensiltelite/Tensile/Tests.

### CODE REVIEW

File: library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/navi31/Equality/navi31_Cijk_Ailk_Bjlk_DB_UserArgs.yaml
Comment (cmingch) [inline library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/navi31/Equality/navi31_Cijk_Ailk_Bjlk_DB_UserArgs.yaml:1]: please provide yaml with WavefrontSize=32 for navi, no matter f32 or f64

File: tensilelite/Tensile/Tests/common/gemm/f32_mac.yaml
Comment (cmingch) [inline tensilelite/Tensile/Tests/common/gemm/f32_mac.yaml:1]: also need to extend BenchmarkProblems with WavefrontSize=32

**Review by cmingch — CHANGES_REQUESTED** [review]

File: clients/include/testing_matmul.hpp [hunk 1]
```diff
@@ -2142,8 +2142,7 @@ void testing_matmul_with_bias(const Arguments& arg)
         static_cast<void>(hipGetDeviceProperties(&deviceProperties, deviceId));
         //workaround before known_bug work
         if(gpu_arch_match(deviceProperties.gcnArchName, "11?")
-           && (arg.gradient || arg.grouped_gemm 
-               || arg.a_type == HIP_R_64F || arg.b_type == HIP_R_64F)) 
+           && (arg.gradient || arg.grouped_gemm))
```
Comment (jichangjichang) [inline clients/include/testing_matmul.hpp:2180]: known bugs should work now? If so, please remove check arch and gradient/grouped_gemm from here.

File: tensilelite/Tensile/Components/MAC_F64.py [hunk 2]
```diff
@@ -45,11 +48,18 @@ def __call__(self, writer, m, innerUnroll):
...
+                    # cStr = ValuC+({a}+{b}*{ThreadTile0})*2
+                    # aStr = ValuA_X{m}_I{iui}+{a}*2
+                    # bStr = ValuB_X{m}_I{iui}+{b}*2
+                    cStr = "ValuC+%d" % ((vars["a"]+vars["b"]*vars["ThreadTile0"])*2)
+                    aStr = "ValuA_X%d_I%d+%d" % (vars["m"], vars["iui"], vars["a"]*2)
+                    bStr = "ValuB_X%d_I%d+%d" % (vars["m"], vars["iui"], vars["b"]*2)
+
+                    module.add(VFmaF64(dst=vgpr(cStr, 2), src0=vgpr(aStr, 2),
+                                       src1=vgpr(bStr, 2), src2=vgpr(cStr, 2)))
+                    module.add(SSetPrior(prior=1, comment="Raise priority while processing macs"))
```
Comment (jichangjichang) [inline tensilelite/Tensile/Components/MAC_F64.py]: should this only raise one time after the first FmaF64 valu?
FmaF64 //first FmaF64
Set_priority 1
FmaF64
....
FmaF64
FmaF64 //last FmaF64
Set_priority 1

File: clients/include/testing_matmul.hpp (hunk 1 above)
Comment (wenchuanchen) [reply clients/include/testing_matmul.hpp:2180]: gradient and grouped_gemm are not yet supported on gfx11xx

**Comment by vin-huang** [comment]
The definition of UseScaleAlphaVec has been changed as below, please kindly modified your YAML file accordingly.

UseScaleAlphaVec=False -> UseScaleAlphaVec=0

UseScaleAlphaVec=True -> UseScaleAlphaVec=1

Redefined UseScaleAlphaVec
0 - Alpha Vector Scaling is disabled
1 - Calculated Alpha Vector Scaling along M-dim
2 - Calculated Alpha Vector Scaling along N-dim
3 - Calculated Alpha Vector Scaling along either M-dim or N-dim.

File: library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/navi33/Equality/navi33_Cijk_Ailk_Bjlk_DB_UserArgs.yaml [hunk 3]
```diff
@@ -0,0 +1,1573 @@
...
+  - - [896, 1024, 1, 32, 896, 896, 896, 1024]
+    - [2, 81.1912]
+  - - [960, 1024, 1, 32, 960, 960, 960, 1024]
+    - [1, 88.5319]
+  - - [1024, 1024, 1, 32, 1024, 1024, 1024, 1024]
+    - [2, 90.0024]
+- null
+- null
+- DeviceEfficiency
+- GridBased
```
Comment (jichangjichang) [inline library/src/amd_detail/rocblaslt/src/Tensile/Logic/asm_full/navi33/Equality/navi33_Cijk_Ailk_Bjlk_DB_UserArgs.yaml]: Equality

**Review by jichangjichang — APPROVED** [review]
lgtm

**Review by cmingch — APPROVED** [review]
LGTM

**Comment by wenchuanchen** [comment]
TODO: modify WavefrontSize=64 to 32 in navi3x f32 yamls

**Pull request merged**